src/azure-cli/azure/cli/command_modules/storage/_params.py

```diff
@@ -357,6 +362,17 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
                    help=sas_help.format(get_permission_help_string(t_blob_permissions)),
                    validator=get_permission_validator(t_blob_permissions))
 
+    with self.argument_context('storage blob restore', resource_type=ResourceType.MGMT_STORAGE) as c:
+        from ._validators import BlobRangeAddAction
+        c.argument('blob_ranges', options_list=['--blob-range', '-r'], action=BlobRangeAddAction, nargs='+',
+                   help='Blob ranges to restore. You need to two values to specify start_range and end_range for each '
```

Comment:
>to [](start = 58, length = 2)

You need to two values -> You need two values

Reply:
since you need two values, why not have parameter of --blob-range-start and --blob-range-end

---
In reply to: [386817455](https://github.com/Azure/azure-cli/pull/12372#discussion_r386817455) [](ancestors = 386817455)

Reply:
We cannot expose two values because the parameter to pass is an array for blob_ranges, which can include multiple blob ranges.